src/test/scala/is/hail/expr/ir/CompileSuite.scala

```diff
@@ -408,4 +407,61 @@ class CompileSuite {
     assert(tOut.isFieldDefined(region, outOff, tOut.fieldIdx("0")))
     assert(region.loadDouble(tOut.loadField(region, outOff, tOut.fieldIdx("0"))) === 8.0)
   }
+
+
+  @Test
+  def testArrayRange() {
+    val tRange = TArray(TInt32())
+    val ir = ArrayRange(In(0, TInt32()), In(1, TInt32()), In(2, TInt32()))
+    val region = Region()
+    val fb = FunctionBuilder.functionBuilder[Region, Int, Boolean, Int, Boolean, Int, Boolean, Long]
+    doit(ir, fb)
+    val f = fb.result()()
+    for {
+      start <- 0 to 2
+      stop <- start to 10
+      step <- 1 to 3
+    } {
+      val aoff = f(region, start, false, stop, false, step, false)
+      val expected = Array.range(start, stop, step)
+      val actual = new UnsafeIndexedSeq(tRange, region, aoff)
+      assert(expected.length == actual.length)
+      assert(actual.sameElements(expected))
+
+      val expected2 = Array.range(stop, start, -step)
+      val aoff2 = f(region, stop, false, start, false, -step, false)
+      val actual2 = new UnsafeIndexedSeq(tRange, region, aoff2)
+      assert(expected2.length == actual2.length)
+      assert(actual2.sameElements(expected2))
+    }
+
+    val (start, stop, step) = (0, Int.MinValue, 1e8.toInt)
```

Comment:
Actually, don't you want something like `(Int.MinValue, Int.MaxValue, 1e8)` here to test the long condition?